lib/lens.ex

```diff
@@ -427,6 +427,35 @@ defmodule Lens do
     lenses |> Enum.reverse() |> Enum.reduce(empty(), &both/2)
   end
 
+  @doc ~S"""
+  Returns a lens that contains what the first lense focuses on, unless it result is empty. In the latter case the
+  result will be what the second lense focuses on.
+
+  This is useful for inserting or updating values.
+
+      iex> upsert = Lens.all() |> Lens.filter(&(&1[:id] == 1)) |> Lens.otherwise(Lens.front())
+      iex> update_in([%{id: 0}, %{id: 1}], [upsert], fn _ -> %{id: 1, x: :y} end)
+      [%{id: 0}, %{id: 1, x: :y}]
+      iex> update_in([%{id: 0}, %{id: 2}], [upsert], fn _ -> %{id: 1, x: :y} end)
+      [%{id: 1, x: :y}, %{id: 0}, %{id: 2}]
+  """
+  @spec otherwise(t, t) :: t
+  deflens_raw otherwise(lens, other_lens) do
+    fn data, fun ->
+      {res, changed} =
+        case get_and_map(lens, data, fun) do
+          {[], _updated} ->
+            {res, updated} = get_and_map(other_lens, data, fun)
+            {[res], updated}
```

Comment:
Why are you wrapping the `res` here into a list in both branches just to `Enum.concat` it at the end?

Reply:
Fair question :)